tools/cmake/FindBpfProg.cmake

```diff
@@ -0,0 +1,188 @@
+# SPDX-License-Identifier: GPL-2.0 OR BSD-3-Clause
+
+#[=======================================================================[.rst:
+FindBpfProg
+--------
+
+Find BpfProg
+
+This module finds if all the dependencies for eBPF Compile-Once-Run-Everywhere
+programs are available and where all the components are located.
+
+The caller may set the following variables to disable automatic
+search/processing for the associated component:
+
+  ``BPFPROG_BPFTOOL_EXE``
+    Path to ``bpftool`` binary
+
+  ``BPFPROG_CLANG_EXE``
+    Path to ``clang`` binary
+
+  ``LIBBPF_INCLUDE_DIRS``
+    Path to ``libbpf`` development headers
+
+  ``LIBBPF_LIBRARIES``
+    Path to `libbpf` library
+
+  ``BPFPROG_VMLINUX_H``
+    Path to ``vmlinux.h`` generated by ``bpftool``. If unset, this module will
+    attempt to automatically generate a copy.
+
+This module sets the following result variables:
+
+::
+
+  BpfProg_FOUND             = TRUE if all components are found
+
+
+This module also provides the ``bpf_prog()`` macro. This macro generates a
+cmake interface library for the BPF program's generated skeleton as well
+as the associated dependencies.
+
+.. code-block:: cmake
+
+  bpf_prog(<name> <source>)
+
+Given an abstract ``<name>`` for a BPF program and the associated ``<source>``
+file, generates an interface library target, ``<name>_skel``, that may be
+linked against by other cmake targets.
+
+Example Usage:
+
+::
+
+  find_package(BpfProg REQUIRED)
+  bpf_prog(myprog myprog.bpf.c)
+  add_executable(myapp myapp.c)
+  target_link_libraries(myapp myprog_skel)
+
+#]=======================================================================]
+
+if(NOT BPFPROG_BPFTOOL_EXE)
+  find_program(BPFPROG_BPFTOOL_EXE NAMES bpftool DOC "Path to bpftool executable")
+endif()
+
+if(NOT BPFPROG_CLANG_EXE)
+  find_program(BPFPROG_CLANG_EXE NAMES clang DOC "Path to clang executable")
+
+  execute_process(COMMAND ${BPFPROG_CLANG_EXE} --version
+    OUTPUT_VARIABLE CLANG_version_output
+    ERROR_VARIABLE CLANG_version_error
+    RESULT_VARIABLE CLANG_version_result
+    OUTPUT_STRIP_TRAILING_WHITESPACE)
+
+  # Check that clang is new enough
+  if(${CLANG_version_result} EQUAL 0)
+    if("${CLANG_version_output}" MATCHES "^clang version ([^\n]+)\n")
+      # Transform X.Y.Z into X;Y;Z which can then be interpreted as a list
+      set(CLANG_VERSION "${CMAKE_MATCH_1}")
+      string(REPLACE "." ";" CLANG_VERSION_LIST ${CLANG_VERSION})
+      list(GET CLANG_VERSION_LIST 0 CLANG_VERSION_MAJOR)
+
+      # Anything older than clang 10 doesn't really work
+      string(COMPARE LESS ${CLANG_VERSION_MAJOR} 10 CLANG_VERSION_MAJOR_LT10)
+      if(${CLANG_VERSION_MAJOR_LT10})
+        message(FATAL_ERROR "clang ${CLANG_VERSION} is too old for BPF CO-RE")
+      endif()
+
+      message(STATUS "Found clang version: ${CLANG_VERSION}")
+    else()
+      message(FATAL_ERROR "Failed to parse clang version string: ${CLANG_version_result}")
```

Comment:
I think this should be `CLANG_version_output`, right ?
Otherwise it just outputs `0`.

Reply:
Yes, nice catch. Thanks